src/community/china.md

```diff
@@ -6,74 +6,171 @@ toc: true
+   ```
+
+1. Add Flutter to your `PATH` environment variable.
+
+   ```terminal
+   $ export PATH="$PWD/flutter/bin:$PATH"
+   ```
+
+1. Run Flutter Doctor to verify your installation.
+
+   ```terminal
+   $ flutter doctor
+   ```
+
+1. Return to the [setting up Flutter]({{site.url}}/get-started/editor)
+   guide and continue from that procedure.
+
+From this example, `flutter pub get` fetches packages from `flutter-io.cn`.
+in any terminal where you set `PUB_HOSTED_URL` and `FLUTTER_STORAGE_BASE_URL`.
+
+Any environment variables set using `export` in this procedure
+only apply to the current terminal window.
+To set these values on a permanent basis,
+add those three `export` commands to the bottom of the `*rc` or `*_profile`
+file that your preferred shell uses.
+
+### Download other versions of Flutter
+
+To install a different version of Flutter found in the
+[archive]({{site.url}}/release/archive),
+replace `storage.googleapis.com` with that of your trusted mirror.
+This should improve download speed.
+
+For example:
 
 * Original URL:<br>
   [`https://storage.googleapis.com/{{path}}`](https://storage.googleapis.com/{{path}})
 
 * Mirrored URL:<br>
   [`https://storage.flutter-io.cn/{{path}}`](https://storage.flutter-io.cn/{{path}})
 
-You must also set two environment variables to upgrade Flutter and use the pub
-package repository in China. Instructions are below.
+## Configure your machine to publish your package
 
-{{site.alert.important}}
-  Use mirror sites only if you _trust_ the provider.
-  The Flutter team cannot verify their reliability or security.
-{{site.alert.end}}
+If you receive errors when attempting to publish your package,
+try one of the following solutions:
 
-## Configuring Flutter to use a mirror site
-
-If you're installing or using Flutter in China,
-it might be helpful to use a trustworthy local
-mirror site that hosts Flutter's dependencies.
-To instruct the Flutter tool to use an alternate storage location,
-you need to set two environment variables, `PUB_HOSTED_URL` and
-`FLUTTER_STORAGE_BASE_URL`, before running the `flutter` command.
-
-Taking macOS or Linux as an example, here are the first few steps in
-the setup process for using a mirror site. Run the following in a Bash
-shell from the directory where you wish to store your local Flutter clone:
-
-```terminal
-$ export PUB_HOSTED_URL=https://pub.flutter-io.cn
-$ export FLUTTER_STORAGE_BASE_URL=https://storage.flutter-io.cn
-$ git clone -b dev {{site.repo.flutter}}.git
-$ export PATH="$PWD/flutter/bin:$PATH"
-$ cd ./flutter
-$ flutter doctor
-```
+  {% comment %}
+  From https://github.com/flutter/website/pull/9338#discussion_r1328077020
+  {% endcomment %}
+
+1. Clear your `PUB_HOSTED_URL` environment variable.
+
+   ```terminal
+   $ unset PUB_HOSTED_URL
+   ```
+
+   Remember to reset this environment variable after you publish.
+
+   ```terminal
+   export PUB_HOSTED_URL=https://pub.flutter-io.cn
+   ```
```

Comment:
**question:** @AlexV525 : Do you have terminal buffer that shows this working that you can share? I'd like to be able to show the successful result of this change.

Reply:
You don't have to reset the env if you're running the publish from a separate window, and this (running from a separate window should be recommended.